lib/cc/kafka/consumer.rb

```diff
@@ -22,15 +22,28 @@ def initialize(client_id, seed_brokers, topic, partition)
         )
       end
 
+      def on_start(&block)
+        @on_start = block
+      end
+
+      def on_stop(&block)
+        @on_stop = block
+      end
+
       def on_message(&block)
         @on_message = block
       end
 
       def start
-        trap(:TERM) { stop }
+        trap(:TERM) do
+          @on_stop.call(@offset) if @on_stop
```

Comment:
No, but i had to move it anyway; see JP's out of date comment for the reasoning.